src/plugins/IFCLoaderPlugin/IFCLoaderPlugin.js

```diff
@@ -0,0 +1,38 @@
+import { Plugin } from "../../viewer/Plugin.js";
+import { GLTFLoaderPlugin } from "../GLTFLoaderPlugin/GLTFLoaderPlugin.js";
+import { ifc2gltf } from "@creooxag/cx-converter";
+
+async function fetchFile(url) {
+    try {
+        const response = await fetch(url);
+        if (!response.ok) {
+            throw new Error(`HTTP error! Status: ${response.status}`);
+        }
+        return response.text();
+    } catch (error) {
+        console.error('Error fetching file:', error);
+    }
+}
+
+class IFCLoaderPlugin extends Plugin {
+    constructor(viewer, cfg = {}) {
+        super("ifcLoader", viewer, cfg);
+    }
+    async load(params = {}) {
+        if (!params.src) {
+            this.error("load() param expected: src");
+        }
+        const data = await fetchFile(params.src);
+        const { gltf, metaData } = await ifc2gltf(data, "remote", params.progressCallback, params.progressTextCallback);
+        const gltfLoader = new GLTFLoaderPlugin(this.viewer);
```

Comment:
Please instantiate the GLTFLoaderPlugin in the constructor and retain as a private property - otherwise it gets instantiated each time we load